internal/agent/device/device.go

```diff
@@ -106,15 +109,26 @@ func NewAgent(
 
 // Run starts the device agent reconciliation loop.
 func (a *Agent) Run(ctx context.Context) error {
-	// orchestrates periodic fetching of device specs and pushing status updates
-	engine := NewEngine(
-		a.fetchSpecInterval,
-		a.syncDeviceSpec,
-		a.statusUpdateInterval,
-		a.statusUpdate,
+	var wg sync.WaitGroup
+	wg.Add(3)
+
+	// Goroutine to handle consoles synchronization
+	go a.syncConsoles(ctx, &wg)
+
+	// Goroutine to handle spec notifier which reads rendered devices from the server
+	notifierEngine := NewEngine(
+		NewUnit(time.Duration(a.fetchSpecInterval), a.specNotifier.PollAndNotify),
 	)
+	go notifierEngine.Run(ctx, &wg)
 
-	return engine.Run(ctx)
+	// Engine to handle device spec and status updates
+	specEngine := NewEngine(
+		NewUnit(time.Duration(a.fetchSpecInterval), a.syncDeviceSpec),
+		NewUnit(time.Duration(a.statusUpdateInterval), a.statusUpdate),
+	)
+	specEngine.Run(ctx, &wg)
+	wg.Wait()
```

Comment:
nit: we should not need a wait group if all goroutines honor ctx. engine.Run can just block?

Reply:
To verify that other goroutines terminated.

Reply:
why would that matter here is the coordination part of some shutdown routine? if yes then maybe we should comment on what we are waiting for.

Reply:
It is cleaner this way.  When the function terminates we know that all the created goroutines were terminated